src/test/daml/Daml/Finance/Instrument/Swap/Test/Currency.daml

```diff
@@ -0,0 +1,97 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.Swap.Test.Currency where
+
+import DA.Date
+import DA.Map qualified as M (fromList)
+import DA.Set (singleton)
+import Daml.Finance.Data.Reference.HolidayCalendar
+import Daml.Finance.Holding.Fungible qualified as Fungible (Factory(..))
+import Daml.Finance.Instrument.Swap.Test.Util
+import Daml.Finance.Interface.Types.Date.Calendar
+import Daml.Finance.Interface.Types.Date.DayCount
+import Daml.Finance.Interface.Types.Date.RollConvention
+import Daml.Finance.Test.Util.Account qualified as Account (createAccount, createFactory, credit)
+import Daml.Finance.Test.Util.Common (createParties)
+import Daml.Finance.Test.Util.Instrument qualified as Instrument (originate)
+import Daml.Script
+
+-- Calculate fix rate payments on a currency swap, including lifecycling and creation of new instrument version.
+run : Script ()
+run = script do
+  [custodian, issuer, investor, calendarDataProvider, publicParty] <-
+    createParties ["Custodian", "Issuer", "Investor", "Calendar Data Provider", "PublicParty"]
+
+  -- Account and holding factory
+  let pp = [("FactoryProvider", singleton publicParty)]
```

Comment:
minor thing: in other scripts we name this variable `fp`, maybe change to that name here (and also in the FloatingRate, ForeignExchange, and InterestRate scripts) to be consistent